python_modules/dagster/dagster/core/host_representation/external_data.py

```diff
@@ -336,13 +345,61 @@ def __new__(cls, error):
         )
 
 
+@whitelist_for_serdes
+class ExternalAssetDependency(
+    namedtuple("_ExternalAssetDependency", "upstream_asset_key input_name")
+):
+    """A definition of a directed edge in the logical asset graph.
+
+    An upstream asset that's depended on, and the corresponding input name in the downstream asset
+    that depends on it.
+    """
+
+    def __new__(cls, upstream_asset_key: AssetKey, input_name: str):
+        return super(ExternalAssetDependency, cls).__new__(
+            cls,
+            upstream_asset_key=upstream_asset_key,
+            input_name=input_name,
+        )
+
+
+@whitelist_for_serdes
+class ExternalAssetDefinition(
```

Comment:
I don't love the name `ExternalAssetDefinition` in the absence of the `asset` decorator.  Also, since this is internal-facing only, maybe something like `ExternalAssetNode`?

Just thinking things through: we can no longer assume there exists at most one asset node per repository / asset_key, right?

Reply:
> Just thinking things through: we can no longer assume there exists at most one asset node per repository / asset_key, right?

I think that's right